adoption/src/main/java/org/dcsa/conformance/standards/adoption/party/AdoptionAdopter.java

```diff
@@ -0,0 +1,100 @@
+import org.dcsa.conformance.core.scenario.ConformanceAction;
+import org.dcsa.conformance.core.state.JsonNodeMap;
+import org.dcsa.conformance.core.toolkit.JsonToolkit;
+import org.dcsa.conformance.core.traffic.ConformanceMessageBody;
+import org.dcsa.conformance.core.traffic.ConformanceRequest;
+import org.dcsa.conformance.core.traffic.ConformanceResponse;
+import org.dcsa.conformance.standards.adoption.AdoptionStandard;
+import org.dcsa.conformance.standards.adoption.action.PutAdoptionStatsAction;
+import org.dcsa.conformance.standards.adoption.action.SupplyScenarioParametersAction;
+
+@Slf4j
+public class AdoptionAdopter extends ConformanceParty {
+
+  public static final String ACTION_ID = "actionId";
+
+  public AdoptionAdopter(
+      String apiVersion,
+      PartyConfiguration partyConfiguration,
+      CounterpartConfiguration counterpartConfiguration,
+      JsonNodeMap persistentMap,
+      PartyWebClient webClient,
+      Map<String, ? extends Collection<String>> orchestratorAuthHeader) {
+    super(
+        apiVersion,
+        partyConfiguration,
+        counterpartConfiguration,
+        persistentMap,
+        webClient,
+        orchestratorAuthHeader);
+  }
+
+  @Override
+  protected void exportPartyJsonState(ObjectNode targetObjectNode) {
+    // no state
+  }
+
+  @Override
+  protected void importPartyJsonState(ObjectNode sourceObjectNode) {
+    // no state
+  }
+
+  @Override
+  protected void doReset() {
+    // no state
+  }
+
+  @Override
+  protected Map<Class<? extends ConformanceAction>, Consumer<JsonNode>> getActionPromptHandlers() {
+    return Map.ofEntries(
+        Map.entry(SupplyScenarioParametersAction.class, this::getAdoptionStats),
+        Map.entry(PutAdoptionStatsAction.class, this::putAdoptionStats));
+  }
+
+  private void getAdoptionStats(JsonNode actionPrompt) {
+    log.info("AdoptionAdopter.getAdoptionStats({})", actionPrompt);
+
+    asyncOrchestratorPostPartyInput(
+        OBJECT_MAPPER
+            .createObjectNode()
+            .put(ACTION_ID, actionPrompt.required(ACTION_ID).asText())
+            .putNull("input"));
+    addOperatorLogEntry(
+        "Provided AdoptionScenarioParameters: %s".formatted(actionPrompt.get(ACTION_ID)));
+  }
+
+  private void putAdoptionStats(JsonNode actionPrompt) {
+    log.info("AdoptionAdopter.putAdoptionStats()");
+
+    String fileName = AdoptionStandard.ADOPTION_STATS_EXAMPLE.formatted(apiVersion);
+    JsonNode jsonResponseBody = JsonToolkit.templateFileToJsonNode(fileName, null);
+    syncCounterpartPut(PutAdoptionStatsAction.PUT_ADOPTION_STATS_URL, jsonResponseBody);
+
+    addOperatorLogEntry("Provided response body, contents of file: %s".formatted(fileName));
+  }
+
+  @Override
+  public ConformanceResponse handleRequest(ConformanceRequest request) {
+    JsonNode jsonResponseBody =
+        JsonToolkit.templateFileToJsonNode(
+            AdoptionStandard.ADOPTION_STATS_EXAMPLE.formatted(apiVersion), Map.ofEntries());
```

Comment:
The example has a hard-coded date and interval, so this response does not respect the request parameters. I would have expected the response to at least be aligned with the interval type and range of the input.

Reply:
Yes, you're right! Well spotted. 

This standard is a little different than others, in a way that the role Adopter and role DCSA are developed. But for the role DCSA, we don't really care much, because we're not going to test our own implementation with it (George told me). So, that's why I didn't spend time in that one.